src/actions/commands/actions.ts

```diff
@@ -671,23 +672,27 @@ class CommandMoveHalfPageDown extends CommandEditorScroll {
       to: this.to,
       by: this.by,
       value: timesToRepeat,
-      revealCursor: false,
+      revealCursor: smoothScrolling ? true : false,
       select:
         [ModeName.Visual, ModeName.VisualBlock, ModeName.VisualLine].indexOf(
           vimState.currentMode
         ) >= 0,
     });
 
-    const newFirstLine = editor.visibleRanges[0].start.line;
-    const newPositionLine = newFirstLine + lineOffset;
-    let newPosition;
+    let newPosition: Position;
+    if (!smoothScrolling) {
+      const newFirstLine = editor.visibleRanges[0].start.line;
```

Comment:
same as `firstLine` var?

Reply:
Same as `newFirstLine` from before this commit ([line `681`](https://github.com/VSCodeVim/Vim/pull/3658/files#diff-ab811cd6f8cfa81bd0cb8223c86e72bfL681) above). `firstLine` is the first line before the scroll, `newFirstLine` is the first line after the scroll.